### PR TITLE
Add repo-launcher extension

### PR DESCRIPTION
**Pull request by gksander**
## Description

This PR proposes a "project launcher" extension (one that I use daily, and decided to abstract for others to use).

The main goal of the extension is to make it easy for developers (who perhaps work in multiple different code repositories) to search for their repos and launch them in various applications, such as their terminal or editor, and to launch various github.com pages for the given repository (such as pulls, commits, issues, etc.). This makes it easy to quickly open and launch the various code projects you're working on in various locations.

Screencast is shown below, but the general flow for this extension is:

- Configure the launcher by opening the settings window.
  - Add as many "containing directories" as needed, these are directories that *contain* your code repositories (might be a `~/Code` dir or something of the likes)
  - Add as many "launch commands" as needed, these are programs or commands to "launch" the repository. E.g., could be the Visual Studio Code program, or a custom command like `pycharm .` which will run a command at the root of the repository.
- (Recommended) Set up an alias for the `Launch Repo` command.
- Search for your repo by typing a few letters.
- Press `Enter` to launch in default program (first in the launch commands list) or `Cmd + K` to choose from the programs you have configured (or in github.com).

## Screencast

https://github.com/raycast/extensions/assets/12721310/8f9aba41-70a6-4e52-91ff-f1509a43d070

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/repo-launcher/src/Settings/SettingsView.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        {[
+          containingDirectories.map((dir) => (
+            <List.Item
+              key={dir}
+              title={dir}
+              icon={Icon.Folder}
+              actions={
+                <ActionPanel>
+                  <Action.ShowInFinder title="Show in Finder" path={dir} />
+                  <Action title={`Remove ${dir}`} onAction={() => removeContainingDirectories(dir)} />
```
Comment (pernielsentikaer) [inline extensions/repo-launcher/src/Settings/SettingsView.tsx:35]:
```suggestion
                  <Action
                    title={`Remove ${dir}`}
                    style={Action.Style.Destructive}
                    icon={Icon.Trash}
                    shortcut={{ modifiers: ["ctrl"], key: "x" }}
                    onAction={() => removeContainingDirectories(dir)}
                  />
```

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Add icons for action panel items

- Add the destructive style/hotkey/icon to all remove actions

- Maybe if nothing is set, push to Settings command (or at lest use List.EmptyView to explain what the user needs to do)
- 

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by gksander** [comment]
@pernielsentikaer thank you for the review, great suggestions! Pushed up a commit with some icons on the action panel items, and added a list-empty view for the project search.

image

**Comment by pernielsentikaer** [comment]
Do you think it would be possible to make a redraw when you return to the list after adding a directory in the settings and returning to the previous view? Otherwise it might be better to try launchCommand instead and send the user back to root 🙂

**Comment by gksander** [comment]
> Do you think it would be possible to make a redraw when you return to the list after adding a directory in the settings and returning to the previous view? Otherwise it might be better to try launchCommand instead and send the user back to root 🙂

Good call @pernielsentikaer – added in an effect to handle this case so that after setting up a containing directory for the first time, we re-run the project-fetch promise!

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**